workflow-gen/Activities/ProcessPaymentActivity.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace WorkflowGen.Activities
+{
+    using System.Threading.Tasks;
+    using Dapr.Client;
+    using Dapr.Workflow;
+    using Microsoft.Extensions.Logging;
+    using WorkflowGen.Models;
+    using System;
+
+    class ProcessPaymentActivity : WorkflowActivity<PaymentRequest, object>
+    {
+        readonly ILogger logger;
+        readonly DaprClient client;
+
+        public ProcessPaymentActivity(ILoggerFactory loggerFactory, DaprClient client)
+        {
+            this.logger = loggerFactory.CreateLogger<ProcessPaymentActivity>();
+            this.client = client;
+        }
+
+        public override async Task<object> RunAsync(WorkflowActivityContext context, PaymentRequest req)
+        {
+            this.logger.LogInformation(
+                "Processing payment: {requestId} for {amount} {item} at ${currency}",
+                req.RequestId,
+                req.Amount,
+                req.ItemBeingPruchased,
+                req.Currency);
+
+            await Task.Delay(TimeSpan.FromSeconds(7));
```

Comment:
Nit: I'd kinda prefer this to be configurable. You could inject it with an `IConfiguration`, or you can ignore this cause it's a test app.